packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { transparentize } from 'polished';
+import facepaint from 'facepaint';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx, keyframes } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+// breakpoints for different screen sizes
+const small = '767px'; // mobile screens, from 0px - 767px
+const medium = '768px'; // tablet screens, from 768px - 1024px
+const large = '1025px'; // laptops/desktop screens, from 1025px and above
+
+export const mq = facepaint([
+  `@media only screen and (max-width: ${small})`,
+  `@media only screen and (min-width: ${medium})`,
+  `@media only screen and (min-width: ${large})`,
+]);
+
+const defaultSpacing = 18;
+
+const fadein = keyframes`
+  from {
+    opacity: 0;
+    transform: translate3d(0, 50px, 0);
+  }
+  to {
+    opacity: 1;
+    transform: translate3d(0, 0px, 0);
+  }
+`;
+
+const backdrop = css`
+  background-color: ${transparentize(0.4, uiColors.black)};
+  overflow-y: auto;
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`;
+
+const scrollContainer = css`
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  bottom: 0;
+  overflow-y: auto;
+  animation: ${fadein} 150ms ease-in-out;
+`;
+
+const modalContentStyle = css`
+  transition: all 200ms ease-in-out;
+  margin: ${defaultSpacing}px auto;
+  padding: 36px;
+  color: ${uiColors.gray.dark3};
+  background-color: ${uiColors.white};
+  border-radius: 3px;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+  position: relative;
+  pointer-events: all;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  small: css`
+    width: 400px;
+  `,
+
+  default: css`
+    width: 720px;
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: absolute;
+  cursor: pointer;
+  right: 12px;
+  top: 12px;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
```

Comment:
Curious why we landed on active to control if the modal is open/visible/shown (other words I've seen used for this)?

Reply:
@DesignerDave  and I actually had a discussion about this: active has been used in other components, so we did it for consistency.